diff --git a/selfdrive/manager/manager.py b/selfdrive/manager/manager.py
index b568fee6..f20b5478 100755
--- a/selfdrive/manager/manager.py
+++ b/selfdrive/manager/manager.py
@@ -74,7 +74,7 @@ def manager_init():
   params.put("GitRemote", get_git_remote(default=""))
 
   # set dongle id
-  reg_res = register(show_spinner=True)
+  reg_res = "00000" #register(show_spinner=True)
   if reg_res:
     dongle_id = reg_res
   else:
