diff --git a/selfdrive/thermald/thermald.py b/selfdrive/thermald/thermald.py
index a2d63f29..c934262d 100755
--- a/selfdrive/thermald/thermald.py
+++ b/selfdrive/thermald/thermald.py
@@ -354,7 +354,7 @@ def thermald_thread():
       set_offroad_alert_if_changed("Offroad_NvmeMissing", (not Path("/data/media").is_mount()))
 
     # Handle offroad/onroad transition
-    should_start = all(startup_conditions.values())
+    should_start = True #all(startup_conditions.values())
     if should_start != should_start_prev or (count == 0):
       params.put_bool("IsOffroad", not should_start)
       HARDWARE.set_power_save(not should_start)
