diff --git a/SConstruct b/SConstruct
index 2e38e8bb..aa9641d1 100644
--- a/SConstruct
+++ b/SConstruct
@@ -50,6 +50,9 @@ if platform.system() == "Darwin":
 if arch == "aarch64" and TICI:
   arch = "larch64"
 
+if platform.system() == "Linux" and arch == "aarch64":
+  arch = "linuxarm64"
+
 USE_WEBCAM = os.getenv("USE_WEBCAM") is not None
 QCOM_REPLAY = arch == "aarch64" and os.getenv("QCOM_REPLAY") is not None
 
@@ -107,7 +110,24 @@ else:
   cxxflags = []
   cpppath = []
 
-  if arch == "Darwin":
+  # ubuntu arm64
+  if arch == "linuxarm64":
+    libpath = [
+      # "#phonelibs/snpe/larch64",
+      "#phonelibs/libyuv/larch64/lib",
+      # "/usr/lib/aarch64-linux-gnu"
+      # "#cereal",
+      "#selfdrive/common",
+      "/usr/lib",
+      "/usr/local/lib",
+    ]
+    #cflags += ["-mcpu=native"]
+    #cxxflags += ["-mcpu=native"]
+    cpppath += [
+      # "/home/dev/mambaforge/envs/op/include",
+      # "/home/dev/mambaforge/envs/op/include/opencv4/"
+    ]
+  elif arch == "Darwin":
     yuv_dir = "mac" if real_arch != "arm64" else "mac_arm64"
     libpath = [
       f"#phonelibs/libyuv/{yuv_dir}/lib",
@@ -136,9 +156,9 @@ else:
     ]
 
   rpath = [
-    "phonelibs/snpe/x86_64-linux-clang",
+    #"phonelibs/snpe/x86_64-linux-clang",
     "cereal",
-    "selfdrive/common"
+    "selfdrive/common",
   ]
 
   # allows shared libraries to work globally
@@ -308,7 +328,7 @@ else:
   qt_dirs += [f"/usr/include/{real_arch}-linux-gnu/qt5/Qt{m}" for m in qt_modules]
 
   qt_libs = [f"Qt5{m}" for m in qt_modules]
-  if arch == "larch64":
+  if arch == "larch64" or arch == "linuxarm64":
     qt_libs += ["GLESv2", "wayland-client"]
   elif arch != "Darwin":
     qt_libs += ["GL"]
